actors/evm/src/interpreter/address.rs

```diff
@@ -31,9 +33,20 @@ impl std::fmt::Debug for EthAddress {
     }
 }
 
+impl TryFrom<&EthAddress> for Address {
+    type Error = fvm_shared::address::Error;
+    fn try_from(addr: &EthAddress) -> Result<Self, Self::Error> {
+        if let Some(addr) = addr.as_id_address() {
+            Ok(addr)
+        } else {
+            Address::new_delegated(EAM_ACTOR_ADDR.id().unwrap(), addr.as_ref())
```

Comment:
By `TryFrom`, I more meant that we needed to handle the case where `EthAddress` cannot be turned into an FVM address. Specifically, it might be in the range 0-255 (a precompile).

Really, we should probably just be unwrapping for the error case here. It's not possible for `new_delegated` to return an error.